### ISSUE TITLE
Rule index ignored with na_ontap_export_policy_rule 

### ISSUE THREAD

**Issue by AdrianBronder** [ansible-collections/netapp.ontap#61]
### Summary

Module na_ontap_export_policy_rule allows providing a "rule_index". But the argument is ignored when the module is executed with REST. Only setting "use_rest" to "never" solves the problem.

### Component Name

netapp.ontap.na_ontap_export_policy_rule

### Ansible Version

```console
Ansible Tower 3.8.0

$ ansible --version
ansible 2.9.27
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.8 (default, Sep  9 2021, 07:49:02) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
```

### ONTAP Collection Version

```console
cat MANIFEST.json
{
 "collection_info": {
  "namespace": "netapp",
  "name": "ontap",
  "version": "21.18.0",
  "authors": [
   "NetApp Ansible Team <ng-ansibleteam@netapp.com>"
  ],
...
```

### ONTAP Version

```console
sridharc-vsim34::> version
NetApp Release 9.9.1P7
```

### Playbook

```console
- name: Create export policy rules
  netapp.ontap.na_ontap_export_policy_rule:
    <<: *login
    state: "{{ object_state }}"
    name: "{{ item.name }}"
    client_match: "{{ item.clients.match }}"
    protocol: "{{ item.protocols }}"
    ro_rule: "{{ item.ro_rule }}"
    rule_index: "{{ item.new_index }}"
    rw_rule: "{{ item.rw_rule }}"
    super_user_security: "{{ item.superuser }}"
    vserver: "{{ item.svm.name }}"
```

### Steps to Reproduce

- set any custom custom value/integer for module parameter "rule_index"

- run the module with "use_rest" set to "always" --> will not set the provided rule_index
- run the module with "use_rest" set to "auto" --> will not set the provided rule_index
- run the module with "use_rest" set to "never" --> will set the provided rule_index properly

### Expected Results

Rule index set in ONTAP as provided in module parameters

### Actual Results

```console
Rule index provided as module parameter is ignored - ONTAP takes a default (next available index)
```

**Comment by lonico** [comment]
The issue is that REST does not support a rule index when creating a rule, but it can be used in a PATCH or DELETE operation.
It does not work well with automation as it is not deterministic.
So the presence of a rule index is not enough to force a ZAPI behavior, as provided with use_rest: never.

**Comment by lonico** [comment]
An idea is to rename the index using
```
curl -d "@test_patch_export_rule.txt" -X PATCH "https://<mgmt-ip>/api/protocols/nfs/export-policies/8589934595/rules/5?new_index=10"
``
 We need to evaluate the feasibility of retrieving the index of a rule when it is created and reassign it to the desired value.

**Comment by lonico** [comment]
We're working on making index a required parameter.

**Comment by carchi8py** [comment]
This is story 4985

**Comment by lonico** [comment]
published as 21.20.0
